include/mxnet/base.h

```diff
@@ -50,6 +50,15 @@
 #endif
 
 /*!
+ *\brief whether to use __builtin_memcpy or not
+ */
+#ifdef MXNET_USE_BUILTIN_MEMCPY
+#define MXNET_MEMCPY __builtin_memcpy
+#else
+#define MXNET_MEMCPY memcpy
+#endif
```

Comment:
Use macro function

Reply:
Is that necessary?

Reply:
Yes, otherwise there's no abstraction of the function signature, which can potentially vary.

Reply:
We should add a mxnet_memcpy function. There is no reason to use macro here.